src/AutoBuild.jl

```diff
@@ -191,6 +192,14 @@ function build_tarballs(ARGS, src_name, src_version, sources, script,
         error("Package name \"$(src_name)\" is not a valid identifier")
     end
 
+    if julia_compat == DEFAULT_JULIA_VERSION_SPEC
+        error("""
+              `julia_compat` must be set explicitly to a valid compat spec string.
+              If this is an existing recipe which didn't previously specify `julia_compat`, use the old default value of `"1.0"`,
+              otherwise for a new recipe you're encouraged to set `julia_compat="1.6"`.
```

Comment:
Should we make this more generic (e.g. say to the current LTS release)? That way we don't need to update the error messages for newer versions.

Reply:
I think sticking to v1.6 is ok, we don't have any particular reason to restrict to newer versions. Maybe I could rephrase slightly to say something along the lines of "at least 1.6"